src/components/Calculator.js

```diff
@@ -1,56 +1,47 @@
-import React, { Component } from 'react';
+import React, { useState } from 'react';
 import calculate from '../logic/calculate';
 
-export default class Calculator extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      obj: {
-        total: '0',
-        next: '',
-        operation: '',
-      },
-    };
-    this.onButtonPress = this.onButtonPress.bind(this);
-  }
+const initial = {
+  total: '0',
+};
 
-  onButtonPress = (obj, btn) => {
-    const newObj = calculate(obj, btn);
+const Calculator = () => {
+  const [obj, setObj] = useState(initial);
+  const onButtonPress = (obj, btnInfo) => {
+    const newObj = calculate(obj, btnInfo);
     if (!newObj.next && !newObj.total) {
-      newObj.next = 0;
+      newObj.next = '0';
     }
-    this.setState({ obj: newObj });
+    setObj(newObj);
   };
+  return (
+    <div className="calc">
+      <p className="calc-screen">
+        {obj.total}
+        {obj.operation}
+        {obj.next}
+      </p>
+      <button onClick={() => onButtonPress(obj, 'AC')} type="button">AC</button>
+      <button onClick={() => onButtonPress(obj, '+/-')} type="button">+/-</button>
+      <button onClick={() => onButtonPress(obj, '%')} type="button" className="">%</button>
+      <button onClick={() => onButtonPress(obj, '÷')} type="button" className="calc-operation">÷</button>
+      <button onClick={() => onButtonPress(obj, '7')} type="button" className="calc-num">7</button>
+      <button onClick={() => onButtonPress(obj, '8')} type="button" className="calc-num">8</button>
+      <button onClick={() => onButtonPress(obj, '9')} type="button" className="calc-num">9</button>
+      <button onClick={() => onButtonPress(obj, 'x')} type="button" className="calc-operation">x</button>
+      <button onClick={() => onButtonPress(obj, '4')} type="button" className="calc-num">4</button>
+      <button onClick={() => onButtonPress(obj, '5')} type="button" className="calc-num">5</button>
+      <button onClick={() => onButtonPress(obj, '6')} type="button" className="calc-num">6</button>
+      <button onClick={() => onButtonPress(obj, '-')} type="button" className="calc-operation">-</button>
+      <button onClick={() => onButtonPress(obj, '1')} type="button" className="calc-num">1</button>
+      <button onClick={() => onButtonPress(obj, '2')} type="button" className="calc-num">2</button>
+      <button onClick={() => onButtonPress(obj, '4')} type="button" className="calc-num">3</button>
+      <button onClick={() => onButtonPress(obj, '+')} type="button" className="calc-operation">+</button>
+      <button onClick={() => onButtonPress(obj, '0')} type="button" className="calc-num calc-zero">0</button>
+      <button onClick={() => onButtonPress(obj, '.')} type="button">.</button>
+      <button onClick={() => onButtonPress(obj, '=')} type="button" className="calc-operation">=</button>
```

Comment:
- [ ] You want to consider giving the right value for the `button 3`. You are giving the value 4 in `onButtonPress` instead of 3. ♻️

Reply:
OMG, how did you find that haha it was so hidden! U got eagle eyes!